core/src/main/java/org/kiwix/kiwixmobile/core/main/AnimatedProgressBar.kt

```diff
@@ -63,31 +54,27 @@ class AnimatedProgressBar : LinearLayout {
    * @param attrs is the attribute set passed by the constructor
    */
   @Suppress("MagicNumber")
-  private fun init(
-    context: Context,
-    attrs: AttributeSet
-  ) {
+  private fun init(context: Context, attrs: AttributeSet) {
     val array = context.theme
-      .obtainStyledAttributes(attrs, styleable.AnimatedProgressBar, 0, 0)
+      .obtainStyledAttributes(attrs, R.styleable.AnimatedProgressBar, 0, 0)
     val backgroundColor: Int
     try {
       // Retrieve the style of the progress bar that the user hopefully set
       val defaultBackgroundColor = 0x424242
       val defaultProgressColor = 0x2196f3
       backgroundColor = array.getColor(
-        styleable.AnimatedProgressBar_backgroundColor, defaultBackgroundColor
+        R.styleable.AnimatedProgressBar_backgroundColor, defaultBackgroundColor
       )
-      mProgressColor = array.getColor(
-        styleable.AnimatedProgressBar_progressColor,
+      progressColor = array.getColor(
+        R.styleable.AnimatedProgressBar_progressColor,
         defaultProgressColor
       )
-      mBidirectionalAnimate =
-        array.getBoolean(styleable.AnimatedProgressBar_bidirectionalAnimate, false)
+      bidirectionalAnimate =
+        array.getBoolean(R.styleable.AnimatedProgressBar_bidirectionalAnimate, false)
     } finally {
       array.recycle()
     }
-    val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
-    inflater.inflate(R.layout.progress_bar, this, true)
+    inflate(context, R.layout.progress_bar, this)
```

Comment:
You might have to import it but this is not using the method in `ViewGroupExtensions`

all you should have to say is `inflate(R.layout.progress_bar)`

Reply:
@macgills Couldn't find it :-( I have searched among existing code too.

Reply:
`import org.kiwix.kiwixmobile.core.extensions.ViewGroupExtensions.inflate`

Reply:
> `import org.kiwix.kiwixmobile.core.extensions.ViewGroupExtensions.inflate`

My bad! Didn't read the previous hint properly.

Thanks! You're helping me to learn a lot!!